app/views/posts/show.html.erb

```diff
@@ -0,0 +1,21 @@
+<section class="d-flex w-100 pt-5 pb-5 justify-content-center">
+  <div class="w-50">
+    <% user = User.where(id: params[:user_id])[0] %>
+    <% if user %>
+      <% post = user.posts.where(id: params[:id])[0] %>
```

Comment:
You can create field variables @user and @post in the controller and if the user or the post doesn't exist we would render 404 page or something of the sort.